server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -220,9 +223,7 @@ static ClusterState addIndexClosedBlocks(final Index[] indices, final ClusterSta
 
         for (IndexMetaData indexToClose : indicesToClose) {
             final Index index = indexToClose.getIndex();
-            if (currentState.blocks().hasIndexBlock(index.getName(), INDEX_CLOSED_BLOCK) == false) {
-                blocks.addIndexBlock(index.getName(), INDEX_CLOSED_BLOCK);
-            }
+            blocks.addIndexBlock(index.getName(), closingBlock);
```

Comment:
what if the index already has a close block? Do we keep on adding these blocks for every close request? Is there a risk here for the cluster state to become very large in case of many close requests?
Should we replace the current close block by a new one or perhaps limit the number of close blocks?

Reply:
Hum. I think that we should always keep the last index closed block only and replace the current close block by a new one.

Reply:
I think we have different options here, with different tradeoffs.
- when starting a close on an index that is not yet fully closed, replace the current closed block by a new one. That means that the possibly ongoing close action cannot successfully complete and the new one will be the one to successfully complete. This is problematic because concurrent calls to close will now make them fail.
- add a new closed block next to the already existing one. Opening will erase all closed blocks. That allows concurrent closes to successfully complete, but possibly adds many blocks to cluster state.
- reuse the existing closed block if there is already one. This combines the advantages of both solutions above.

Reply:
>     * reuse the existing closed block if there is already one. This combines the advantages of both solutions above.

It is indeed more elegant but if multiple close actions share the same unique block the actions become kind of dependant of each others, no? If the first response received is a failure (let's say, a shard is relocating and in INITIALIZING state) the index block should be removed, but we could have potential responses with the same block that arrive later and they won't be able to succeed.